### ISSUE TITLE
Support für Dreame L10s Ultra

### ISSUE THREAD

**Issue by hardlinerap** [iobroker-community-adapters/ioBroker.mihome-vacuum#674]
Infos zum Gerät:

-   Marke: Dreame
-   Modell: L10s Ultra
-   Bezeichnung: dreame.vacuum.r2228o

Model dreame.vacuum.r2228o not supported! Please open issue on git: https://github.com/iobroker-community-adapters/ioBroker.mihome-vacuum/issues

Vielen Dank!

**Comment by thoml95** [comment]
Für HASS gibt es neuerdings eine Dreame Integration, vielleicht hilft das ja.

https://github.com/Tasshack/dreame-vacuum

**Comment by stony2k** [comment]
Ja genau, die Integration nutze ich aktuell. Natürlich wäre es mir eine Integration im ioBroker lieber :)

**Comment by dirkhase** [comment]
Would be also nice for me. Can I support?

**Comment by DemigodCode** [comment]
Hey, bin gerade dabei. Sollte die nächsten Tage, wenn nicht beim PR irgendwas auftaucht erreichbar sein.
Allerdings erstmal nur rudimentärer Basic Support. Je nach Modell gibt es "Sonderfunktionen" für die die Objekte im Adapter erweitert werden müssten und dann die von der Basisklasse abgeleitet andere speziellere Klassen erstellt werden müssen.
Wie zum Beispiel für diesen hier.

Ich kann das auf Wunsch später alles nachziehen mit ein bisschen Zeit, aber erstmal besser als gar nix ;-)

**Comment by stony2k** [comment]
> Hey, bin gerade dabei. Sollte die nächsten Tage, wenn nicht beim PR irgendwas auftaucht erreichbar sein. Allerdings erstmal nur rudimentärer Basic Support. Je nach Modell gibt es "Sonderfunktionen" für die die Objekte im Adapter erweitert werden müssten und dann die von der Basisklasse abgeleitet andere speziellere Klassen erstellt werden müssen. Wie zum Beispiel für diesen hier.
> 
> Ich kann das auf Wunsch später alles nachziehen mit ein bisschen Zeit, aber erstmal besser als gar nix ;-)

Cool ist dein PR im aktuellen 3.9.0 release drin? Warte noch, dass ich im ioBroker das Update des Adapters sehe.

**Comment by dirkhe** [comment]
Nein, der dream Support wird erstmal nur im 4.0. angelegt und kommt noch nicht nach ioBroker, siehe https://github.com/iobroker-community-adapters/ioBroker.mihome-vacuum/discussions/731

**Comment by xXBJXx** [comment]
`dreame.vacuum.r2228o` has been listed in the supported-devices table of the README for some time (basic control and cleaning history through the Dreame library), so I am closing this request as completed. If something does not work on 6.1.0 (Node.js 22.13 or newer, js-controller 7.2.2 or newer and Admin 7.8.23 or newer), please open a new issue with the debug log and the `deviceInfo.model` / `deviceInfo.fw_ver` values. Room, zone and map functions for Dreame models are tracked in #823.
